sim/paladin/sor.go

```diff
@@ -7,173 +7,148 @@ import (
+	improvedSoR := paladin.improvedSoR()
+
+	for i, rank := range ranks {
+		i, rank := i, rank
+		if paladin.Level < rank.level {
+			break
+		}
+
+		/*
+		 * Seal of Righteousness is a Spell/Aura that when active makes the paladin capable of procing
+		 * two different SpellIDs depending on a paladin's casted spell or melee swing.
+		 *
+		 * (Judgement of Righteousness):
+		 *   - Deals flat damage that is affected by Improved SoR talent, and
+		 *     has a spellpower scaling that is unaffected by that talent.
+		 *   - Targets magic defense and rolls to hit and crit.
+		 *
+		 * (Seal of Righteousness):
+		 *   - Procs from white hits.
+		 *   - Cannot miss or be dodged/parried/blocked if the underlying white hit lands.
+		 *   - Deals damage that is a function of weapon speed, and spellpower.
+		 *   - Has 0.85 scale factor on base damage if using 1h, 1.2 if using 2h.
+		 *   - CANNOT CRIT.
+		 */
+
+		minDamage := rank.judge.minDamage + rank.judge.scale*float64(min(paladin.Level, rank.scaleLevel)-rank.level)
+		maxDamage := rank.judge.maxDamage + rank.judge.scale*float64(min(paladin.Level, rank.scaleLevel)-rank.level)
+
+		judgeSpell := paladin.RegisterSpell(core.SpellConfig{
+			ActionID:    core.ActionID{SpellID: rank.judge.spellID},
+			SpellSchool: core.SpellSchoolHoly,
+			DefenseType: core.DefenseTypeMagic,
+			ProcMask:    core.ProcMaskEmpty,
+			Flags:       core.SpellFlagMeleeMetrics,
+
+			BonusCritRating: paladin.holyCrit(), // TODO check whether fanaticism applies
+
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+			BonusCoefficient: rank.judge.coeff,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				baseDamage := sim.Roll(minDamage, maxDamage) * improvedSoR
+				spell.CalcAndDealDamage(sim, target, baseDamage, spell.OutcomeMagicHitAndCrit)
 			},
-		},
+		})
 
-		ApplyEffects: func(sim *core.Simulation, _ *core.Unit, _ *core.Spell) {
-			paladin.ApplySeal(aura, onJudgementProc, sim)
-		},
-	})
-}
+		value := rank.proc.value + rank.proc.scale*float64(min(paladin.Level, rank.scaleLevel)-rank.level)
 
-func (paladin *Paladin) registerSealOfRighteousnessSpellAndAura() {
-	paladin.SealOfRighteousness = make([]*core.Spell, sorRanks+1)
-	paladin.SealOfRighteousnessAura = make([]*core.Aura, sorRanks+1)
+		damage := value / 100 * core.TernaryFloat64(paladin.Has2hEquipped(), 1.2, 0.85) * paladin.MainHand().SwingSpeed
 
-	for rank := 1; rank <= sorRanks; rank++ {
-		if int(paladin.Level) >= sorLevels[rank] {
-			paladin.MaxRankRighteousness = rank
-			paladin.applySealOfRighteousnessSpellAndAuraBaseConfig(rank)
-		}
+		procSpell := paladin.RegisterSpell(core.SpellConfig{
+			ActionID:    core.ActionID{SpellID: rank.proc.spellID},
+			SpellSchool: core.SpellSchoolHoly,
+			DefenseType: core.DefenseTypeMagic,
+			ProcMask:    core.ProcMaskEmpty,
+			Flags:       core.SpellFlagMeleeMetrics,
+
+			DamageMultiplier: 1, // TODO check whether this benefits from weapon specializations
```

Comment:
The general rule with weapon specialization talents is that despite what the Mod damage done (phys) effect may imply, anything that targets the melee defense type while the relevant weapon is equipped will benefit.

This means that SoR procs and a bunch of other holy damage abilities benefit, and this has been confirmed with testing. Please re-add any weapon specialization multipliers that were removed from holy abilities.